load-code/pe/translate-characteristic-into-permission.yml

```diff
@@ -0,0 +1,22 @@
+rule:
+  meta:
+    name: translate characteristic into permission
+    namespace: load-code/pe
+    author: "@Ana06"
+    scope: function
+    examples:
+      - E4C33AC3638EEF68311F8AC0D72483C7:0x401480
+  features:
+    - and:
```

Comment:
Introducing a `match at least X` scope would be helpful here, as we could write something like that:
```
- match at least 3:
      - and:
	    - number: 0x40000000 = IMAGE_SCN_MEM_READ
	    - number: 0x2 = PAGE_READONLY
	  - and:
	    - number: 0x20000000 = IMAGE_SCN_MEM_EXECUTE
        - number: 0x10 = PAGE_EXECUTE
      - and:
        - number: 0x60000000 = IMAGE_SCN_MEM_READ | IMAGE_SCN_MEM_EXECUTE
        - number: 0x20 = PAGE_EXECUTE_READ
      - and:
        - number: 0xC0000000 = IMAGE_SCN_MEM_READ | IMAGE_SCN_MEM_WRITE
        - number: 0x4 = PAGE_READWRITE
      - and:
        - number: 0xE0000000 = IMAGE_SCN_MEM_READ | IMAGE_SCN_MEM_WRITE | IMAGE_SCN_MEM_EXECUTE
        - number: 0x40 = PAGE_EXECUTE_READWRITE
      - optional:
        - number: 0x1 = PAGE_NOACCESS

```

I know I can write an equivalent rule already but it is not a very beautiful one.

I can open an issue for this one as well.

Reply:
does `3 or more` not work here?

Reply:
i think it should

Reply:
I don't think so as we only do that for `count(<feature>)...` but not across multiple features?

Reply:
It could be called `3 or more`, but still I think this is something we don't currently support. 🤔

Reply:
i dont think so, i think `3 or more` is parsed like `optional` which supports arbitrary children terms:
https://github.com/fireeye/capa/blob/master/capa/rules.py#L329-L331

let me pull up the explorer and make a test case

Reply:
![image](https://user-images.githubusercontent.com/156560/112532215-2add5780-8d6e-11eb-8188-c97b71e4e611.png)

Reply:
i agree there's also a term like `count(mnem(mov)): 3 or more` but that's separate

Reply:
btw that was soooo easy with the explorer @mike-hunhoff

Reply:
🙈 wow, this needs to be documented better

Reply:
You are right @williballenthin, it works! 😮 although `optional` doesn't work as I expected. The following two seem to be equivalent:

```
- 2 or more: 
	- number: 0x1
	- number: 0x2
	- number: 0x3
	- optional:
		- number: 0x4
```

```
- 2 or more: 
	- number: 0x1
	- number: 0x2
	- number: 0x3
	- number: 0x4
```

but I had actually expected this behaviour:
```
- and:
	- 2 or more: 
		  - number: 0x1
		  - number: 0x2
		  - number: 0x3
	- optional:
		- number: 0x4
```

Reply:
`optional` always evaluates to true which has been a point of confusion before